spec/facility_spec.rb

```diff
@@ -23,4 +25,186 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+  describe '#register_vehicle' do
+    before(:each) do
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+      @camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice} )
+    end
+
+    it 'cannot register a vehicle if the service is not present at the facility' do
+      expect(@facility.services).to eq []
+      expect(@facility.register_vehicle(@cruz)). to eq nil
+    end
+
+    it 'can register a vehicle' do
+      @facility.add_service('Vehicle Registration')
+
+      @facility.register_vehicle(@cruz)
+
+      expect(@facility.registered_vehicles).to eq [@cruz]
+
+      @facility.register_vehicle(@bolt)
+
+      expect(@facility.registered_vehicles).to eq [@cruz, @bolt]
+    end
+
+    it 'can collect a fee when registering a vehicle' do
+      @facility.add_service('Vehicle Registration')
+      
+      @facility.register_vehicle(@cruz)
+      
+      expect(@facility.collected_fees).to eq 100
+    end
+
+    it 'can assigns the proper plate when registering a vehicle' do
+      @facility.add_service('Vehicle Registration')
+
+      @facility.register_vehicle(@cruz)
+      @facility.register_vehicle(@bolt)
+      @facility.register_vehicle(@camaro)
+
+      expect(@cruz.plate_type).to eq :regular
+      expect(@bolt.plate_type).to eq :ev
+      expect(@camaro.plate_type).to eq :antique
+    end
+  end 
+
+  describe '#collect_fee' do
+    before(:each) do
+      @facility.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+      @camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice} )
+    end
+
+    it 'can collect a fee (regular)' do
```

Comment:
Didn't we already have a test for collecting the fees on line 58? No need for the one on line 58 if you have all these ones here. 
I like that you have tested all possible values from this method